### PR TITLE
game cards revised

### PR DESCRIPTION
**Pull request by yzemaze**
+ player names truncated (max. 19 char), full name on hover
+ Max Players / Created inline
+ Updated right aligned
+ time when today, else date
+ fulldate on hover
+ kick buttons hover text
+ commas separated from name
+ no commas between buttons
+ margins/paddings optimized
+ b => strong, u => em

### CODE REVIEW

File: models/base.rb [hunk 1]
```diff
@@ -7,7 +7,23 @@ def pp_created_at
     created_at&.strftime('%Y-%m-%d')
   end
 
+  def pp_created_time
```
Comment (tobymao) [inline models/base.rb:10]: let's not do this. get rid of all of these and just return created_at and updated_at as unix time stamps, time.to_i or something like that, and just render the time on the client, i updated chat to do the same

Comment (yzemaze) [reply models/base.rb:10]: so just 2 defs there or scrap it altogether?

**Comment by tobymao** [comment]
refactor everything to use the proper time stamps

File: assets/app/view/game_card.rb [hunk 2]
```diff
@@ -148,55 +147,84 @@ def render_link(href, click, text)
...
       ]
 
       if new?
-        children << h(:div, [h(:b, 'Max Players: '), @gdata['max_players']])
-        children << h(:div, [h(:b, 'Created: '), @gdata['created_at']])
+        children << h(:div, { style: { display: 'inline' } }, [h(:strong, 'Max Players: '), @gdata['max_players']])
+        children << h(:div, { style: { display: 'inline', float: 'right' } }, [
+          h(:strong, 'Created: '),
+          h(:span, { attrs: { title: Time.at(@gdata['created_at']).strftime('%F %T') } },
+            if Time.at(@gdata['created_at']).strftime('%F') == Time.now.strftime('%F')
```
Comment (tobymao) [inline assets/app/view/game_card.rb:193]: this and the other stuff can be cleaned up

created_at = @gdata['created_at']

created_at >= Time.now - 86400 # could make this a function maybe

File: assets/app/view/game_card.rb [hunk 3]
```diff
@@ -148,55 +147,84 @@ def render_link(href, click, text)
...
-        h(:div, [h(:b, 'Players: '), *p_elm]),
+        h(:div, [h(:strong, 'Id: '), @gdata['id'].to_s]),
+        h(:div, [h(:strong, 'Description: '), @gdata['description']]),
+        h(:div, [h(:strong, 'Players: '), *p_elm]),
       ]
 
       if new?
-        children << h(:div, [h(:b, 'Max Players: '), @gdata['max_players']])
-        children << h(:div, [h(:b, 'Created: '), @gdata['created_at']])
+        children << h(:div, { style: { display: 'inline' } }, [h(:strong, 'Max Players: '), @gdata['max_players']])
```
Comment (tobymao) [inline assets/app/view/game_card.rb:189]: don't you have css class inline?

File: assets/app/view/game_card.rb [hunk 4]
```diff
@@ -148,55 +147,84 @@ def render_link(href, click, text)
...
           .map { |k, v| "#{k} (#{v})" }
           .join(', ')
 
         children << h(:div, [
-          h(:b, 'Result: '),
+          h(:strong, 'Result: '),
           result,
         ])
-      else
+      elsif @gdata['round'] || @gdata['updated_at']
```
Comment (tobymao) [inline assets/app/view/game_card.rb:209]: can you do elsif gdata['round'] 
elsif gdata['updated_at']

also i don't like this style of putting if end inside of an array

File: models/base.rb [hunk 5]
```diff
@@ -4,10 +4,10 @@
 
 class Base
   def pp_created_at
```
Comment (tobymao) [inline models/base.rb:6]: change the name of this because it's no longer pretty printed

File: assets/app/view/game_card.rb (hunk 3 above)
Comment (yzemaze) [reply assets/app/view/game_card.rb:189]: O tempora! O mores! toby pushes CSS ;)
You did create it for inline-block (yay for blame). Will change.

File: assets/app/view/game_card.rb (hunk 4 above)
Comment (yzemaze) [reply assets/app/view/game_card.rb:209]: If updated_at is only set when round is set, it’s even simpler.

File: assets/app/view/game_card.rb (hunk 2 above)
Comment (yzemaze) [reply assets/app/view/game_card.rb:193]: note sure about 86400 as at least 1 hour less might be better to not run into overlap issues, but yes

File: public/assets/main.css [hunk 6]
```diff
@@ -158,7 +159,12 @@ label.right {
   display: inline-block;
 }
 
-.inline { display: inline-block }
+.inline {
+  display: inline
```
Comment (tobymao) [inline public/assets/main.css:163]: this seems like a dangerous change because all other inline elements were doing inline-block

Comment (yzemaze) [reply public/assets/main.css:163]: It seems, until you grep for "\.inline". It returned only assets/app/view/game/issue_shares.rb. There I changed it to .inline-block. All other inline-blocks are defined inline.

**Pull request merged**